libraries/abstractions/common_io/test/test_iot_i2c.c

```diff
@@ -449,6 +454,82 @@ TEST( TEST_IOT_I2C, AFQP_IotI2CWriteThenReadAsyncSuccess )
 
 /*-----------------------------------------------------------*/
 
+/**
+ * @brief Test Function for I2C async read success
+ */
+TEST( TEST_IOT_I2C, AFQP_IotI2CReadAsyncAssisted )
+{
+    IotI2CHandle_t xI2CHandle;
+    int32_t lRetVal;
+    uint8_t ucReadBuf[16];
+    char cMsg[50]={0};
+
+    IotI2CConfig_t xI2CConfig = {
+        .ulBusFreq = testIotI2C_BAUDRATE,
+        .ulMasterTimeout =  testIotI2C_DEFAULT_TIMEOUT
+    };
+
+    if(TEST_PROTECT())
+    {
+        /* Open i2c to initialize hardware */
+        if (gIotI2cHandle[testIotI2C_INSTANCE] == NULL)
+        {
+            xI2CHandle = iot_i2c_open(testIotI2C_INSTANCE);
+            TEST_ASSERT_NOT_EQUAL(NULL, xI2CHandle);
+        }
+        else
+        {
+            xI2CHandle = gIotI2cHandle[testIotI2C_INSTANCE];
+        }
+
+        /* Set completion callback */
+        iot_i2c_set_callback(xI2CHandle, prvI2CCallback, NULL );
+
+        /* Set i2c congifuration */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetMasterConfig, &xI2CConfig);
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Set i2c slave address for writing the device register */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetSlaveAddr, &ucAssistedTestIotI2CSlaveAddr );
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Set i2c not stop between transaction */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSendNoStopFlag, NULL);
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Write the device register address */
+        lRetVal = iot_i2c_write_sync(xI2CHandle, &xtestIotI2CDeviceRegister, sizeof(xtestIotI2CDeviceRegister));
+        TEST_ASSERT_EQUAL(IOT_I2C_SUCCESS, lRetVal);
+
+        /* Set i2c slave address to read from the device register */
+        lRetVal = iot_i2c_ioctl(xI2CHandle, eI2CSetSlaveAddr, &ucAssistedTestIotI2CSlaveAddr );
```

Comment:
Does your implementation remember the slave address?

Reply:
It depends on the ll hal. Normally it should not be remembered by hal layer.

Reply:
This is a separate discussion from this PR(not need change now). 

Firstly it doesn't depend on the low level driver, we can surely do it ourselves.
Either remembering or not is OK to me. But if **normally** it should not be remembered, our API should be designed as write(handle, address, buffer, bufferSize).

Reply:
I think I misunderstood before. It is definitely remembered before i2c_close.

Reply:
ok, if the I2C implementation in your code base is also that way, you can delete those extra lines of "set address" in this PR.

Reply:
Set slave address should be enforced before a new read or write. Does not matter whether the salve addr is changed or remembered. The consistency should be kept to not cause confusion.